sympy/core/mul.py

```diff
@@ -549,7 +549,11 @@ def _handle_for_oo(c_part, coeff_sign):
 
         # 0
         elif coeff is S.Zero:
-            # we know for sure the result will be 0
+            # we know for sure the result will be 0 except the multiplicand
+            # is infinity
+            if(len(c_part) > 0):
+                if c_part[0].is_commutative and c_part[0].is_finite == False:
```

Comment:
Oh, I see.  I agree with @asmeurer, you should use `is_finite == False`.  #9227 - is wrong, sorry for misguiding from my side.